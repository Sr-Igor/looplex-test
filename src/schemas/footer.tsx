// import { IFooterProps } from 'components/Footer';

import github from '../../public/img/github-logo.png';
import linkedin from '../../public/img/linkedin-logo.png';

export const schema = [
  {
    text: 'Github',
    url: 'https://github.com/Sr-Igor',
    image: github
  },
  {
    text: 'Â© 2023 - Todos os direitos reservados'
  },
  {
    text: 'Linkedin',
    url: 'https://www.linkedin.com/in/igor-rezende-375b231b5/',
    image: linkedin
  }
];
