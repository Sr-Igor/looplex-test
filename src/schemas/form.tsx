import { customizeValidator } from '@rjsf/validator-ajv8';
import ptLocalizer from 'ajv-i18n/localize/pt-BR';

import { RJSFSchema } from '@rjsf/utils';

export const schema: RJSFSchema = {
  type: 'object',
  required: ['email', 'name', 'birthday'],
  properties: {
    avatar: { type: 'string', title: 'Avatar' },
    email: {
      type: 'string',
      title: 'Email do cliente',
      format: 'email',
      maxLength: 50,
      minLength: 5
    },
    name: { type: 'string', title: 'Nome Completo', minLength: 3 },
    birthday: { type: 'integer', title: 'Data de nascimento' }
  }
};

export const uiSchema = {
  avatar: {
    'ui:widget': 'fileField',
    'ui:placeholder': 'Avatar',
    'ui:classNames': 'avatar-field'
  },
  email: {
    'ui:widget': 'textField',
    'ui:options': {
      extra: 'Insira um email válido',
      icon: 'email'
    },

    'ui:inputProps': {
      placeholder: 'Email',
      inputMode: 'email'
    }
  },
  name: {
    'ui:widget': 'textField',
    'ui:options': {
      extra: 'Insira o nome do usuário',
      icon: 'user'
    },

    'ui:inputProps': {
      placeholder: 'Nome'
    }
  },
  birthday: {
    'ui:widget': 'dateField',
    'ui:inputProps': {
      placeholder: 'Data de nascimento',
      inputMode: 'date'
    }
  }
};

// eslint-disable-next-line @typescript-eslint/no-explicit-any
export const validator = customizeValidator({}, ptLocalizer as any);
