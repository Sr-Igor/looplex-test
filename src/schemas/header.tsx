import logo from '../../public/assets/logo.png';

export const schema = {
  logo,
  title: 'Deal Gran',
  do_login: 'Fa√ßa login',
  panel: 'Painel'
};
