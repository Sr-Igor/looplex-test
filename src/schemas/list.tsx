export const schema = {
  title: 'Gestão de Usuários',
  new: 'Novo Usuário',
  empty: 'Ops! Você não possui nenhum usuário registrado'
};
