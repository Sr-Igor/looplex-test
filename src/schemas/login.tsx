import { GithubOutlined, GoogleOutlined } from '@ant-design/icons';

export const schema = {
  google: 'Entrar com Google',
  googleIcon: <GoogleOutlined />,
  github: 'Entrar com Github',
  githubIcon: <GithubOutlined />,
  outLogin: 'Entrar sem autenticação',
  outLoginPlaceholder: 'Digite seu nome para entrar',
  enter: 'Entrar'
};
