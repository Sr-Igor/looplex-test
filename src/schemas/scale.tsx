import logo from '../../public/assets/logo.png';

export const schema = {
  logo,
  do_login: 'Faça login',
  panel: 'Ver Painel',
  title: 'Gerencie seus usuários',
  subtitle: 'Cada desafio a menos é um passo para o sucesso',
  drops: [
    {
      title: 'Organização',
      timer: 1000,
      drop: 7000
    },
    {
      title: 'Velocidade',
      timer: 1200,
      drop: 6000
    },
    {
      title: 'Fluidez',
      timer: 1400,
      drop: 5000
    },
    {
      title: 'Simplicidade',
      timer: 1600,
      drop: 4000
    },
    {
      title: 'Elegância',
      timer: 1800,
      drop: 3000
    },
    {
      title: 'Intuitivo',
      timer: 2000,
      drop: 2000
    }
  ]
};
