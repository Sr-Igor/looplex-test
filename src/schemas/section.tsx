import {
  PlusOutlined,
  BookOutlined,
  CalendarOutlined
} from '@ant-design/icons';

export const schema = [
  {
    icon: <PlusOutlined />,
    title: 'Benefício 1',
    description:
      'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only.'
  },
  {
    icon: <CalendarOutlined />,
    title: 'Benefício 2',
    description:
      'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only.'
  },
  {
    icon: <BookOutlined />,
    title: 'Benefício 3',
    description:
      'Lorem Ipsum is simply dummy text of the printing and typesetting industry. Lorem Ipsum has been the industry standard dummy text ever since the 1500s, when an unknown printer took a galley of type and scrambled it to make a type specimen book. It has survived not only.'
  }
];
