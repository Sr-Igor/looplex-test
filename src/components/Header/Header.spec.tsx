import { schema } from 'schemas/header';

import { render, screen } from 'utils/tests-utils';

import { Header } from '.';

jest.mock('next/navigation', () => ({
  useRouter: jest.fn(() => ({ push: jest.fn() }))
}));

jest.mock('next-auth/react', () => ({
  useSession: jest.fn(() => ({})),
  signOut: jest.fn()
}));

describe('<Main />', () => {
  it('should render the component', () => {
    const { container } = render(
      <Header onPressLogin={() => console.log('teste')} schema={schema} />
    );

    expect(screen.getByText(schema.do_login)).toBeInTheDocument();

    expect(container).toMatchSnapshot();
  });
});
