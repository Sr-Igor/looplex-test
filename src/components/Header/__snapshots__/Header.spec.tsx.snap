// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main /> should render the component 1`] = `
.c0 {
  max-width: 130rem;
  width: 100%;
  height: 100px;
  background-color: #0D0D0D;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: justify;
  -webkit-justify-content: space-between;
  -ms-flex-pack: justify;
  justify-content: space-between;
  padding: 0 2.4rem;
  position: fixed;
  top: 0;
  z-index: 20;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 1.6rem;
  cursor: pointer;
}

.c2 {
  width: 80px;
  height: auto;
}

.c4 {
  color: #fff;
  font-size: 2.4rem;
  font-weight: 600;
  text-transform: uppercase;
}

.c6 {
  height: 80px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  gap: 1.6rem;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: none;
  background-color: transparent;
}

.c6:hover .c3 {
  color: #F1C40F;
}

.c6:disabled {
  cursor: default;
}

.c7 {
  width: 40px;
  height: 40px;
  border-radius: 50%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  background-color: #F1C40F;
}

.c8 {
  font-size: 1.6rem;
  font-weight: 600;
  color: #f0efeb;
  -webkit-transition: all 0.3s ease-in-out;
  transition: all 0.3s ease-in-out;
}

.c5 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  gap: 1.6rem;
}

@media (max-width:450px) {
  .c2 {
    width: 40px;
  }
}

@media (max-width:450px) {
  .c4 {
    font-size: 1.2rem;
  }
}

@media (max-width:450px) {
  .c7 {
    width: 20px;
    height: 20px;
  }
}

<div>
  <header
    class="c0"
  >
    <div
      class="c1"
    >
      <img
        alt="Logo"
        class="c2"
        data-nimg="1"
        decoding="async"
        height="40"
        loading="lazy"
        src="/_next/image?url=%2Fimg.jpg&w=96&q=75"
        srcset="/_next/image?url=%2Fimg.jpg&w=48&q=75 1x, /_next/image?url=%2Fimg.jpg&w=96&q=75 2x"
        style="color: transparent;"
        width="40"
      />
      <h3
        class="c3 c4"
      >
        Deal Gran
      </h3>
    </div>
    <div
      class="c5"
    >
      <button
        class="c6"
      >
        <div
          class="c7"
        >
          <span
            aria-label="user"
            class="anticon anticon-user"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="user"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M858.5 763.6a374 374 0 00-80.6-119.5 375.63 375.63 0 00-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 00-80.6 119.5A371.7 371.7 0 00136 901.8a8 8 0 008 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 008-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
          </span>
        </div>
        <p
          class="c8"
        >
          Fa√ßa login
        </p>
      </button>
    </div>
  </header>
</div>
`;
